### PR TITLE
Reduce flow overhead by avoiding transforms when possible

### PR DESCRIPTION
**Pull request by peppy**
This needs some regression testing, but the overhead for the average `FlowContainer` was just too high previously. If we want to support potentially changing `AutoSizeDuration` at runtime then further logic is also required. Consider this a proposal.

### CODE REVIEW

**Comment by Tom94** [comment]
The problem with this approach is that previously, existing position-based transforms would have been interrupted whereas now they are not.

This, however, raises the question whether manually setting `Position` shouldn't _by itself_ interrupt existing transforms. I would say so, but an implementation would require adding a separate code path for transforms to update `Position` without interrupting themselves.

A short-term fix would be to call `ClearTransforms("Position")` or similar prior to updating `Position`.

**Comment by peppy** [comment]
Yeah, that's the case I mentioned in the description.

**Comment by Tom94** [comment]
I see. However, also consider that somebody might transform the position of a child independently of the layouting happening. Could, for instance, happen when implementing draggable elements within a flow.

**Comment by smoogipoo** [comment]
Changing the position of children inside a `FlowContainer` should be an invalid operation anyway. Doing so will re-invalidate the flow and re-set the position.

File: osu.Framework/Graphics/Containers/FlowContainer.cs [hunk 1]
```diff
@@ -189,7 +189,15 @@ private void performLayout()
 
                 var finalPos = positions[i];
                 if (d.Position != finalPos)
-                    d.TransformTo(d.PopulateTransform(new FlowTransform { Rewindable = false }, finalPos, LayoutDuration, LayoutEasing));
+                {
+                    if (LayoutDuration > 0)
+                        d.TransformTo(d.PopulateTransform(new FlowTransform { Rewindable = false }, finalPos, LayoutDuration, LayoutEasing));
+                    else
+                    {
+                        ClearTransforms(false, nameof(Position));
```
Comment (smoogipoo) [inline osu.Framework/Graphics/Containers/FlowContainer.cs]: This should be `d.ClearTransforms(...)` instead.

Comment (smoogipoo) [reply osu.Framework/Graphics/Containers/FlowContainer.cs]: Requires a test, this would've been caught with it.

Comment (peppy) [reply osu.Framework/Graphics/Containers/FlowContainer.cs]: should tie in with @ekrctb's change too; he's already done half the work for adding tests to cover this.

**Comment by peppy** [comment]
Tests added.

**Pull request merged**